### PR TITLE
Update samples to .NET 7, fix asset path to be relative

### LINKED ISSUE #155: Calling a page with routing parameters causes the map to not initialize correctly
If a GeoBlazor map is on a page that uses routing to pass in a parameter, the map is not initialized correctly.  Invoking a map before using route parameters will allow the map to display correctly.  Calling the page without passing in the parameter as part of the route allows the map to initialize correctly. Passing in parameters as query strings also works correctly.

I have updated the demo from a previous issue to demonstrate the problem. The repo is https://github.com/anotherlab/geoblazer-issue-2-marker-offsets. You will need to add your own Esri API key and that information is included in the README.md file.

## What is happening ##
When a page is invoked with a parameter passed in as part of the route, no map details are displayed and numerous Javascript errors can be seen in the browser console log

## What should happen ##
The map should be displayed.

## Steps to replicate the problems ##
1. Clone or get the latest version of the [demo](https://github.com/anotherlab/geoblazer-issue-2-marker-offsets)
2. Restore the nuget packages
3. Edit [MauiProgram.cs](https://github.com/anotherlab/GeoBlazor-issue-1/blob/main/demo1/MauiProgram.cs#L22) and replace "INSERT-YOUR-ESRI-API-KEY-HERE" with your API key
4. Build and the run app on the platform of your choice
5. Click on the link labeled "Route param". This must be the first page invoked.

When viewed with Chrome Inspect the following Javascript errors come up in the log
<details>
<summary>Error log</summary

```
arcGisJsInterop.js:52          GET https://0.0.0.0/pageroute/student/_content/dymaptic.GeoBlazor.Core/assets/esri/widgets/Popup/t9n/Popup_en.json 404 (Not found)
arcGisJsInterop.js:52          GET https://0.0.0.0/pageroute/student/_content/dymaptic.GeoBlazor.Core/assets/esri/t9n/common_en.json 404 (Not found)
arcGisJsInterop.js:52          GET https://0.0.0.0/pageroute/student/_content/dymaptic.GeoBlazor.Core/assets/esri/widgets/Attribution/t9n/Attribution_en.json 404 (Not found)
arcGisJsInterop.js:52          GET https://0.0.0.0/pageroute/student/_content/dymaptic.GeoBlazor.Core/assets/esri/widgets/Zoom/t9n/Zoom_en.json 404 (Not found)
arcGisJsInterop.js:52          GET https://0.0.0.0/pageroute/student/_content/dymaptic.GeoBlazor.Core/assets/esri/widgets/ScaleBar/t9n/ScaleBar_en.json 404 (Not found)
arcGisJsInterop.js:52          GET https://0.0.0.0/pageroute/student/_content/dymaptic.GeoBlazor.Core/assets/esri/widgets/Popup/t9n/Popup.json 404 (Not found)
arcGisJsInterop.js:52          GET https://0.0.0.0/pageroute/student/_content/dymaptic.GeoBlazor.Core/assets/esri/t9n/common.json 404 (Not found)
arcGisJsInterop.js:52          GET https://0.0.0.0/pageroute/student/_content/dymaptic.GeoBlazor.Core/assets/esri/widgets/Attribution/t9n/Attribution.json 404 (Not found)
arcGisJsInterop.js:52          GET https://0.0.0.0/pageroute/student/_content/dymaptic.GeoBlazor.Core/assets/esri/widgets/Zoom/t9n/Zoom.json 404 (Not found)
arcGisJsInterop.js:52          GET https://0.0.0.0/pageroute/student/_content/dymaptic.GeoBlazor.Core/assets/esri/widgets/Compass/t9n/Compass_en.json 404 (Not found)
arcGisJsInterop.js:52          GET https://0.0.0.0/pageroute/student/_content/dymaptic.GeoBlazor.Core/assets/esri/widgets/ScaleBar/t9n/ScaleBar.json 404 (Not found)
arcGisJsInterop.js:74 wasm streaming compile failed: TypeError: Failed to execute 'compile' on 'WebAssembly': Incorrect response MIME type. Expected 'application/wasm'.
arcGisJsInterop.js:52          GET https://0.0.0.0/pageroute/student/_content/dymaptic.GeoBlazor.Core/assets/esri/widgets/Compass/t9n/Compass.json 404 (Not found)
arcGisJsInterop.js:1 [esri.widgets.Popup] widget-intl:locale-error esri.widgets.Popup pe {name: 'intl:message-bundle-error', details: {…}, message: 'Errors occurred while loading "esri/widgets/Popup/t9n/Popup"'}
arcGisJsInterop.js:1 [esri.widgets.Popup] widget-intl:locale-error esri.widgets.Popup pe {name: 'intl:message-bundle-error', details: {…}, message: 'Errors occurred while loading "esri/t9n/common"'}
arcGisJsInterop.js:1 [esri.widgets.Attribution] widget-intl:locale-error esri.widgets.Attribution pe {name: 'intl:message-bundle-error', details: {…}, message: 'Errors occurred while loading "esri/widgets/Attribution/t9n/Attribution"'}
arcGisJsInterop.js:1 [esri.widgets.Zoom] widget-intl:locale-error esri.widgets.Zoom pe {name: 'intl:message-bundle-error', details: {…}, message: 'Errors occurred while loading "esri/widgets/Zoom/t9n/Zoom"'}
arcGisJsInterop.js:1 [esri.widgets.ScaleBar] widget-intl:locale-error esri.widgets.ScaleBar pe {name: 'intl:message-bundle-error', details: {…}, message: 'Errors occurred while loading "esri/widgets/ScaleBar/t9n/ScaleBar"'}
arcGisJsInterop.js:1 [esri.widgets.Compass] widget-intl:locale-error esri.widgets.Compass pe {name: 'intl:message-bundle-error', details: {…}, message: 'Errors occurred while loading "esri/widgets/Compass/t9n/Compass"'}
arcGisJsInterop.js:52 TypeError: Cannot destructure property 'zoomIn' of 'this.messages' as it is null.
    at o.render (arcGisJsInterop.js:10571:172038)
    at s (arcGisJsInterop.js:52:18720)
    at ZA (arcGisJsInterop.js:1:666288)
    at render (arcGisJsInterop.js:52:19121)
    at p (arcGisJsInterop.js:29:27361)
    at Object.merge (arcGisJsInterop.js:29:27973)
    at o._attach (arcGisJsInterop.js:52:21240)
    at _internalHandles.add.Ea.initial (arcGisJsInterop.js:52:21173)
    at arcGisJsInterop.js:28:96162
    at w3t (arcGisJsInterop.js:28:96244)
arcGisJsInterop.js:52 TypeError: Cannot read properties of null (reading 'mi')
    at o._renderLine (arcGisJsInterop.js:10581:44897)
    at o.render (arcGisJsInterop.js:10581:44020)
    at s (arcGisJsInterop.js:52:18720)
    at ZA (arcGisJsInterop.js:1:666288)
    at render (arcGisJsInterop.js:52:19121)
    at p (arcGisJsInterop.js:29:27361)
    at Object.merge (arcGisJsInterop.js:29:27973)
    at o._attach (arcGisJsInterop.js:52:21240)
    at _internalHandles.add.Ea.initial (arcGisJsInterop.js:52:21173)
    at arcGisJsInterop.js:28:96162
arcGisJsInterop.js:52 TypeError: Cannot read properties of null (reading 'reset')
    at o.render (arcGisJsInterop.js:10571:164726)
    at s (arcGisJsInterop.js:52:18720)
    at ZA (arcGisJsInterop.js:1:666288)
    at render (arcGisJsInterop.js:52:19121)
    at p (arcGisJsInterop.js:29:27361)
    at Object.merge (arcGisJsInterop.js:29:27973)
    at o._attach (arcGisJsInterop.js:52:21240)
    at _internalHandles.add.Ea.initial (arcGisJsInterop.js:52:21173)
    at arcGisJsInterop.js:28:96162
    at w3t (arcGisJsInterop.js:28:96244)
arcGisJsInterop.js:1 [esri.core.workers.WorkerOwner] ErrorEvent {isTrusted: true, message: "Uncaught NetworkError: Failed to execute 'importSc…sri/core/workers/RemoteClient.js' failed to load.", filename: 'blob:https://0.0.0.0/b0d0b20b-9aec-4bd5-b858-aa47c64f7a59', lineno: 1, colno: 4171, …}
arcGisJsInterop.js:1 [esri.core.workers.WorkerOwner] ErrorEvent {isTrusted: true, message: "Uncaught NetworkError: Failed to execute 'importSc…sri/core/workers/RemoteClient.js' failed to load.", filename: 'blob:https://0.0.0.0/b0d0b20b-9aec-4bd5-b858-aa47c64f7a59', lineno: 1, colno: 4171, …}
arcGisJsInterop.js:1 [esri.core.workers.WorkerOwner] ErrorEvent {isTrusted: true, message: "Uncaught NetworkError: Failed to execute 'importSc…sri/core/workers/RemoteClient.js' failed to load.", filename: 'blob:https://0.0.0.0/b0d0b20b-9aec-4bd5-b858-aa47c64f7a59', lineno: 1, colno: 4171, …}
arcGisJsInterop.js:10571 Uncaught (in promise) TypeError: Cannot destructure property 'zoomIn' of 'this.messages' as it is null.
    at o.render (arcGisJsInterop.js:10571:172038)
    at s (arcGisJsInterop.js:52:18720)
    at ZA (arcGisJsInterop.js:1:666288)
    at render (arcGisJsInterop.js:52:19121)
    at p (arcGisJsInterop.js:29:27361)
    at Object.merge (arcGisJsInterop.js:29:27973)
    at o._attach (arcGisJsInterop.js:52:21240)
    at _internalHandles.add.Ea.initial (arcGisJsInterop.js:52:21173)
    at arcGisJsInterop.js:28:96162
    at w3t (arcGisJsInterop.js:28:96244)
arcGisJsInterop.js:10581 Uncaught (in promise) TypeError: Cannot read properties of null (reading 'mi')
    at o._renderLine (arcGisJsInterop.js:10581:44897)
    at o.render (arcGisJsInterop.js:10581:44020)
    at s (arcGisJsInterop.js:52:18720)
    at ZA (arcGisJsInterop.js:1:666288)
    at render (arcGisJsInterop.js:52:19121)
    at p (arcGisJsInterop.js:29:27361)
    at Object.merge (arcGisJsInterop.js:29:27973)
    at o._attach (arcGisJsInterop.js:52:21240)
    at _internalHandles.add.Ea.initial (arcGisJsInterop.js:52:21173)
    at arcGisJsInterop.js:28:96162
arcGisJsInterop.js:10571 Uncaught (in promise) TypeError: Cannot read properties of null (reading 'reset')
    at o.render (arcGisJsInterop.js:10571:164726)
    at s (arcGisJsInterop.js:52:18720)
    at ZA (arcGisJsInterop.js:1:666288)
    at render (arcGisJsInterop.js:52:19121)
    at p (arcGisJsInterop.js:29:27361)
    at Object.merge (arcGisJsInterop.js:29:27973)
    at o._attach (arcGisJsInterop.js:52:21240)
    at _internalHandles.add.Ea.initial (arcGisJsInterop.js:52:21173)
    at arcGisJsInterop.js:28:96162
    at w3t (arcGisJsInterop.js:28:96244)
```
<details>

Tested under Windows 11 with Windows and Android as target platforms. Compiled with VS 2022 17.5.3 and GeoBlazor 2.0.2-Beta 2

### PR DESCRIPTION
**Pull request by TimPurdum**
Closes #155. Also updates samples projects to target .NET 7.

After discussion, I removed the `.` in the asset path and tested all the sample applications. They still functioned correctly in WASM, MAUI, and Server. I also added a reminder in the `assetFiles.md` docs page about setting the "base" path in html.

### CODE REVIEW

File: samples/dymaptic.GeoBlazor.Core.Sample.Maui/dymaptic.GeoBlazor.Core.Sample.Maui.csproj [hunk 1]
```diff
@@ -27,48 +27,55 @@
...
+        <MauiFont Include="Resources\Fonts\*" />
 
         <!-- Raw Assets (also remove the "Resources\Raw" prefix) -->
-        <MauiAsset Include="Resources\Raw\**" LogicalName="%(RecursiveDir)%(Filename)%(Extension)"/>
+        <MauiAsset Include="Resources\Raw\**" LogicalName="%(RecursiveDir)%(Filename)%(Extension)" />
     </ItemGroup>
 
     <ItemGroup>
-        <MauiImage Include="Resources\Splash\geoblazor.png"/>
+      <Content Remove="appsettings.json" />
```
Comment (seahro) [inline samples/dymaptic.GeoBlazor.Core.Sample.Maui/dymaptic.GeoBlazor.Core.Sample.Maui.csproj:50]: I will need to remember this, will we need a appsettings.development.json here also?

**Review by seahro — APPROVED** [review]
looks good, mostly version updates.  I will need to get geoblazor up on my mac and start testing it there

File: samples/dymaptic.GeoBlazor.Core.Sample.Maui/dymaptic.GeoBlazor.Core.Sample.Maui.csproj (hunk 1 above)
Comment (TimPurdum) [reply samples/dymaptic.GeoBlazor.Core.Sample.Maui/dymaptic.GeoBlazor.Core.Sample.Maui.csproj:50]: It's not necessary, unless we wanted one. I just wanted one file to demonstrate that appsettings do work for MAUI.

**Comment by TimPurdum** [comment]
> looks good, mostly version updates. I will need to get geoblazor up on my mac and start testing it there

@seahro this would be awesome if you could test MAUI iOS and MAUI Mac deployments for us at some point!

**Pull request merged**